### ISSUE TITLE
Chest + hoppers do not work on 1.18-rc3

### ISSUE THREAD

**Issue by Cubxity** [PaperSpigot/Paper#6941]
### Expected behavior

Items from chest should transfer to the hoppers beneath it.

### Observed/Actual behavior

```
[19:19:47 ERROR]: BlockEntity threw exception at world:122,64,241
smp-minecraft-1  | java.lang.AbstractMethodError: Receiver class net.minecraft.world.level.block.entity.TileEntityHopper does not define or inherit an implementation of the resolved method 'abstract net.minecraft.world.level.World W()' of interface net.minecraft.world.level.block.entity.IHopper.
smp-minecraft-1  | 	at net.minecraft.world.level.block.entity.HopperBlockEntity.hopperPull(HopperBlockEntity.java:244) ~[?:?]
smp-minecraft-1  | 	at net.minecraft.world.level.block.entity.HopperBlockEntity.lambda$suckInItems$3(HopperBlockEntity.java:462) ~[?:?]
smp-minecraft-1  | 	at net.minecraft.world.level.block.entity.HopperBlockEntity.anyMatch(HopperBlockEntity.java:440) ~[?:?]
smp-minecraft-1  | 	at net.minecraft.world.level.block.entity.HopperBlockEntity.suckInItems(HopperBlockEntity.java:459) ~[?:?]
smp-minecraft-1  | 	at net.minecraft.world.level.block.entity.HopperBlockEntity.lambda$pushItemsTick$0(HopperBlockEntity.java:141) ~[?:?]
smp-minecraft-1  | 	at net.minecraft.world.level.block.entity.HopperBlockEntity.tryMoveItems(HopperBlockEntity.java:163) ~[?:?]
smp-minecraft-1  | 	at net.minecraft.world.level.block.entity.HopperBlockEntity.pushItemsTick(HopperBlockEntity.java:140) ~[?:?]
smp-minecraft-1  | 	at net.minecraft.world.level.chunk.LevelChunk$BoundTickingBlockEntity.tick(LevelChunk.java:1165) ~[?:?]
smp-minecraft-1  | 	at net.minecraft.world.level.chunk.LevelChunk$RebindableTickingBlockEntityWrapper.tick(LevelChunk.java:1117) ~[?:?]
smp-minecraft-1  | 	at net.minecraft.world.level.Level.tickBlockEntities(Level.java:953) ~[?:?]
smp-minecraft-1  | 	at net.minecraft.server.level.ServerLevel.tick(ServerLe
vel.java:676) ~[?:?]
smp-minecraft-1  | 	at net.minecraft.server.MinecraftServer.tickChildren(MinecraftServer.java:1605) ~[paper-1.18-rc3.jar:git-Paper-"6498480"]
smp-minecraft-1  | 	at net.minecraft.server.dedicated.DedicatedServer.tickChildren(DedicatedServer.java:480) ~[paper-1.18-rc3.jar:git-Paper-"6498480"]
smp-minecraft-1  | 	at net.minecraft.server.MinecraftServer.tickServer(MinecraftServer.java:1469) ~[paper-1.18-rc3.jar:git-Paper-"6498480"]
smp-minecraft-1  | 	at net.minecraft.server.MinecraftServer.runServer(MinecraftServer.java:1263) ~[paper-1.18-rc3.jar:git-Paper-"6498480"]
smp-minecraft-1  | 	at net.minecraft.server.MinecraftServer.lambda$spin$0(MinecraftServer.java:316) ~[paper-1.18-rc3.jar:git-Paper-"6498480"]
smp-minecraft-1  | 	at java.lang.Thread.run(Thread.java:833) ~[?:?]
[19:19:47 ERROR]: BlockEntity threw exception at world:122,64,241
smp-minecraft-1  | java.lang.AbstractMethodError: Receiver class net.minecraft.world.level.block.entity.TileEntityHopper does not define or inherit an implementation of the resolved method 'abstract net.minecraft.world.level.World W()' of interface net.minecraft.world.level.block.entity.IHopper.
smp-minecraft-1  | 	at net.minecraft.world.level.block.entity.HopperBlockEntity.hopperPull(HopperBlockEntity.java:244) ~[?:?]
smp-minecraft-1  | 	at net.minecraft.world.level.block.entity.HopperBlockEntity.lambda$suckInItems$3(HopperBlockEntity.java:462) ~[?:?]
smp-minecraft-1  | 	at net.minecraft.world.level.block.entity.HopperBlockEntity.anyMatch(HopperBlockEntity.java:440) ~[?:?]
smp-minecraft-1  | 	at net.minecraft.world.level.block.entity.HopperBlockEntity.suckInItems(HopperBlockEntity.java:459) ~[?:?]
smp-minecraft-1  | 	at net.minecraft.world.level.block.entity.HopperBlockEntity.lambda$pushItemsTick$0(HopperBlockEntity.java:141) ~[?:?]
smp-minecraft-1  | 	at net.minecraft.world.level.block.entity.HopperBlockEntity.tryMoveItems(HopperBlockEntity.java:163) ~[?:?]
smp-minecraft-1  | 	at net.minecraft.world.level.block.entity.HopperBlockEntity.pushItemsTick(HopperBlockEntity.java:140) ~[?:?]
smp-minecraft-1  | 	at net.minecraft.world.level.chunk.LevelChunk$BoundTickingBlockEntity.tick(LevelChunk.java:1165) ~[?:?]
smp-minecraft-1  | 	at net.minecraft.world.level.chunk.LevelChunk$RebindableTickingBlockEntityWrapper.tick(LevelChunk.java:1117) ~[?:?]
smp-minecraft-1  | 	at net.minecraft.world.level.Level.tickBlockEntities(Level.java:953) ~[?:?]
smp-minecraft-1  | 	at net.minecraft.server.level.ServerLevel.tick(ServerLevel.java:676) ~[?:?]
smp-minecraft-1  | 	at net.minecraft.server.MinecraftServer.tickChildren(MinecraftServer.java:1605) ~[paper-1.18-rc3.jar:git-Paper-"6498480"]
smp-minecraft-1  | 	at net.minecraft.server.dedicated.DedicatedServer.tickChildren(DedicatedServer.java:480) ~[paper-1.18-rc3.jar:git-Paper-"6498480"]
smp-minecraft-1  | 	at net.minecraft.server.MinecraftServer.tickServer(MinecraftServer.java:1469) ~[paper-1.18-rc3.jar:git-Paper-"6498480"]
smp-minecraft-1  | 	at net.minecraft.server.MinecraftServer.runServer(MinecraftServer.java:1263) ~[paper-1.18-rc3.jar:git-Paper-"6498480"]
smp-minecraft-1  | 	at net.minecraft.server.MinecraftServer.lambda$spin$0(MinecraftServer.java:316) ~[paper-1.18-rc3.jar:git-Paper-"6498480"]
smp-minecraft-1  | 	at java.lang.Thread.run(Thread.java:833) ~[?:?]
```

### Steps/models to reproduce

![image](https://user-images.githubusercontent.com/27609129/143719657-ec36c43f-b5d3-4729-b05f-5c56c6d4e775.png)

### Plugin and Datapack List

Plugins: 
- Chunky
- LuckPerms
- spark
- UnifiedMetrics
- VanishNoPacket

### Paper version

git-Paper-"6498480" (MC: 1.18 Release Candidate 3)

**Comment by lynxplay** [comment]
Replicable :+1: 
Re-obfuscation obfuscates  
`net.minecraft.world.level.block.entity.Hopper#getLevel` to  
`net.minecraft.world.level.block.entity.IHopper#W`  
while  
`net.minecraft.world.level.block.entity.BlockEntity#getLevel` is obfuscated to  
`net.minecraft.world.level.block.entity.TileEntity#k`.

**Comment by e-im** [comment]
Fixed by 4eb45ccac9699e6dbd56bd1a062340910c51e014
